src/ScoreSystem.js

```diff
@@ -0,0 +1,27 @@
+
+export class ScoreSystem {
+	constructor() {
+  }
+
+	initialize(player) {
+		player.score = 0;
```

Comment:
Although tempting because of how easy it is, you shouldn't add properties to an object after its construction. It prevents the JIT from performing certain optimisations.